src/dropzone/DropzoneDirective.ts

```diff
@@ -53,6 +54,17 @@ export class DropzoneDirective implements ng.IDirective {
       });
     }
 
+    // dropzone cancel
```

Comment:
Add comments to elaborate on the override.

Reply:
Thanks for pointing this out. This comment has been updated to indicate it is overriding the dropzone cancel function and calling the backend's cancelUpload.

Since we call removeFile after the file successfully uploads (to remove it from the dropzone container), this function always gets called both when the user cancels the file and when it uploads successfully. Therefore, we check to see if the file has actually uploaded fully or not before calling cancelUpload.